### PR TITLE
Optimized Intimidate's code

### PR DESCRIPTION
**Pull request by LOuroboros**
## Description
Remember how I optimized Bad Dreams a few days ago *(#2434)* and mentioned how Intimidate was similarly complex even though it didn't need to?
I decided to fix that because I wanted to so that's another battle script command freed to use moving forward.
As far as I tested, everything's working normally, but more tests are always welcomed.

Before:
https://user-images.githubusercontent.com/4485172/201494549-5f1e810f-95a5-4a27-ab8c-d73297852b49.mp4

After *(this PR)*:
https://user-images.githubusercontent.com/4485172/201494560-f1a46ee3-8627-4d54-8ab3-73cb0b1c71d4.mp4

"*But... doesn't that look exactly the same?*"
I know! Isn't it awesome? Less code to deal with which means manipulating or copying Intimidate is super easy now! teddiawwrsa

## **Discord contact info**
Lunos#4026

### CODE REVIEW

File: data/battle_scripts_1.s [hunk 1]
```diff
@@ -8220,27 +8216,30 @@ BattleScript_IntimidateActivatesLoop:
...
 	setgraphicalstatchangevalues
 	playanimation BS_TARGET, B_ANIM_STATS_CHANGE, sB_ANIM_ARG1
 	printstring STRINGID_PKMNCUTSATTACKWITH
 	waitmessage B_WAIT_TIME_LONG
-	call BattleScript_TryAdrenalineOrb
-BattleScript_IntimidateActivatesLoopIncrement:
+@ Perform stat nerf
+BattleScript_IntimidateLoopIncrement:
 	addbyte gBattlerTarget, 1
-	goto BattleScript_IntimidateActivatesLoop
```
Comment (ghoulslash) [inline data/battle_scripts_1.s:8232]: needs to check adrenaline orb if intimidate is successful, too

**Review by ghoulslash — CHANGES_REQUESTED** [review]

File: data/battle_scripts_1.s (hunk 1 above)
Comment (LOuroboros) [reply data/battle_scripts_1.s:8232]: Done :eyes:

File: data/battle_scripts_1.s [hunk 2]
```diff
@@ -8220,27 +8216,30 @@ BattleScript_IntimidateActivatesLoop:
...
 .endif
-	statbuffchange STAT_CHANGE_NOT_PROTECT_AFFECTED | STAT_CHANGE_ALLOW_PTR, BattleScript_IntimidateActivatesLoopIncrement
-	jumpifbyte CMP_GREATER_THAN, cMULTISTRING_CHOOSER, 1, BattleScript_IntimidateActivatesLoopIncrement
+BattleScript_IntimidateEffect:
+	statbuffchange STAT_CHANGE_NOT_PROTECT_AFFECTED | MOVE_EFFECT_CERTAIN, NULL
 	setgraphicalstatchangevalues
 	playanimation BS_TARGET, B_ANIM_STATS_CHANGE, sB_ANIM_ARG1
 	printstring STRINGID_PKMNCUTSATTACKWITH
 	waitmessage B_WAIT_TIME_LONG
 	call BattleScript_TryAdrenalineOrb
```
Comment (ghoulslash) [inline data/battle_scripts_1.s:8225]: `BattleScript_TryAdrenalineOrb` needs BS_SCRIPTING for its string (and maybe more), so before calling it, we need `copybyte sBATTLER, gBattlerTarget`

**Review by ghoulslash — CHANGES_REQUESTED** [review]

File: data/battle_scripts_1.s (hunk 2 above)
Comment (LOuroboros) [reply data/battle_scripts_1.s:8226]: Done :eyes:

I never asked about it, I take it that `sBATTLER` acts as a 1-byte big shortcut to `BS_SCRIPTING` then :eyes:

Comment (ghoulslash) [reply data/battle_scripts_1.s:8226]: `sBATTLER` is the address in EWRAM of `gBattleScripting.battler` whereas `BS_SCRIPTING` is a caseId to load gBattleScripting.battler in `GetBattlerForBattleScript`

File: include/battle.h [hunk 3]
```diff
@@ -158,7 +158,6 @@ struct SpecialStatus
     u8 statLowered:1;
     u8 lightningRodRedirected:1;
     u8 restoredBattlerSprite: 1;
-    u8 intimidatedMon:1;
```
Comment (AsparagusEduardo) [inline include/battle.h:161]: Forgot to submit this comment that I had pending:
> Change to an unused name variation, as there's `// End of byte` below.

**Pull request merged**